### PR TITLE
feat(molecules): add SfScrollable component

### PR DESCRIPTION
**Pull request by aniamusial**
# Related issue
#523 

# Scope of work
Scrollable component with toggleable button

# Checklist

- [x] I followed [composition rules](https://docs.storefrontui.io/contributing/coding-guidelines.html#component-rules) for my component
- [x] I tested the component in most common device sizes (can be tested in Storybook [from viewport addon in top menu](https://github.com/storybooks/storybook/tree/master/addons/viewport))

### CODE REVIEW

**Comment by filrak** [comment]
I think it needs a little bit more stretching and should be done via JS. CSS support is really bad https://caniuse.com/#feat=css-scrollbar

File: packages/vue/src/components/molecules/SfScrollable/SfScrollable.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        this.$refs.content.$el.style.setProperty("max-height", "unset");
+        this.$refs.content.$el.style.setProperty("overflow-y", "unset");
+      } else {
+        this.$refs.content.$el.style.setProperty(
+          "max-height",
+          this.maxContentHeight
+        );
+      }
+    }
+  }
```
Comment (EduardDopler) [inline packages/vue/src/components/molecules/SfScrollable/SfScrollable.js]: Could you check if you can convert this logic to a `computed` property?
I think `handleClick` should only set `isHidden` (can even be done inside the template), the rest may work with only computed properties.

**Comment by EduardDopler** [comment]
Hi @aniamusial The problem with the test here is that you try to import a CSS module in the JS part of the component. Seems like our Jest config isn't prepared for CSS imports in JS files. But shouldn't CSS imports go into the component's SCSS file anyway? @pspaczek

**Comment by pspaczek** [comment]
Yes, we should import CSS in sass file :)

File: packages/vue/src/components/molecules/SfScrollable/SfScrollable.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import simplebar from "simplebar-vue";
+import "simplebar/dist/simplebar.min.css";
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfScrollable/SfScrollable.js]:
```suggestion
```

File: packages/shared/styles/components/SfScrollable.scss [hunk 3]
```diff
@@ -0,0 +1,24 @@
+@import "../variables";
```
Comment (pspaczek) [inline packages/shared/styles/components/SfScrollable.scss]:
```suggestion
@import "~simplebar/dist/simplebar.min.css";
@import "../variables";
```

**Comment by aniamusial** [comment]
@pspaczek clearing cache and moving to .scss file worked like a charm :) Thank you :clap: !

File: packages/vue/src/components/molecules/SfScrollable/SfScrollable.js (hunk 1 above)
Comment (aniamusial) [reply packages/vue/src/components/molecules/SfScrollable/SfScrollable.js]: Moved it to a watcher and `isHidden` is now handled inside the template

File: packages/shared/styles/components/SfScrollable.scss [hunk 4]
```diff
@@ -0,0 +1,25 @@
+@import "~simplebar/dist/simplebar.min.css";
+@import "../variables";
+
+$scrollable__max-width: 13.5rem !default;
```
Comment (pspaczek) [inline packages/shared/styles/components/SfScrollable.scss]:
```suggestion
```
@aniamusial, why we limit width inside component, probably parent, should have responsibility for that

File: packages/shared/styles/components/SfScrollable.scss [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+.sf-scrollable {
+    --content-height: 2rem;
```
Comment (pspaczek) [inline packages/shared/styles/components/SfScrollable.scss]:
```suggestion
    --content-height: 10rem;
```
we can set the highest value

File: packages/vue/src/components/molecules/SfScrollable/SfScrollable.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      default: ""
+    },
+    closeText: {
+      type: String,
+      default: ""
+    }
+  },
+  watch: {
+    isHidden(value) {
+      if (!value) {
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfScrollable/SfScrollable.js]: not for SSR <3 watch immediate run on SSR :) and you use $refs :) 
meybe? 
```
<simplebar class="sf-scrollable__content" :style="{maxHeight, overflowY}">
```

File: packages/vue/src/components/molecules/SfScrollable/SfScrollable.stories.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+            label: "Navy"
+          },
+          {
+            color: "pink",
+            label: "Pink"
+          }
+        ]
+      };
+    },
+    template: `<SfScrollable show-text="View all colors" close-text="Hide" max-content-height="6rem">
```
Comment (pspaczek) [inline packages/vue/src/components/molecules/SfScrollable/SfScrollable.stories.js]: @aniamusial what we should do when max-content-heigh === content-height, we should disable scroll and enable it if the height is biggest

File: packages/vue/src/components/molecules/SfScrollable/SfScrollable.js (hunk 6 above)
Comment (pspaczek) [reply packages/vue/src/components/molecules/SfScrollable/SfScrollable.js]: and can we use `Simplebar` not `simplebar`? :)

File: packages/vue/src/components/molecules/SfScrollable/SfScrollable.js (hunk 1 above)
Comment (EduardDopler) [reply packages/vue/src/components/molecules/SfScrollable/SfScrollable.js]: Super cool 👍

File: packages/vue/src/components/molecules/SfScrollable/SfScrollable.stories.js [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+  .add("Content height by CSS", () => ({
+    components: { SfScrollable, SfProductOption },
+    props: {
+      showText: {
+        default: text("showText", "View all colors", "Props")
+      },
+      hideText: {
+        default: text("hideText", "Hide colors", "Props")
+      },
+      maxContentHeight: {
```
Comment (aniamusial) [inline packages/vue/src/components/molecules/SfScrollable/SfScrollable.stories.js]: Just a suggestion - can we hide maxContentHeight prop for "Content Height by CSS" ? IMO it's a bit misleading.

Comment (pspaczek) [reply packages/vue/src/components/molecules/SfScrollable/SfScrollable.stories.js]: 👍 you have right <3

**Pull request merged**